### PR TITLE
Use AppLookupObject to get handle_metadata in `.from_id()`

### PR DESCRIPTION
**Pull request by freider**
Also rename _initialize_from_proto to _initialize_from_handle_metadata (long name, but a bit clearer?)

### CODE REVIEW

**Comment by erikbern** [comment]
Why is the `WhichOneOf` and `getattr` needed?

Great if you can add tests too!

**Comment by freider** [comment]
> Why is the `WhichOneOf` and `getattr` needed?
> 
> Great if you can add tests too!

Mostly future proofing - the returned handle_metadata (previously object) can be any handle's proto object, but is currently a one_of that only has function as a valid field. When/if we add handle_metadata for any other object, this will make it automatically pick the "one_of" that is set. Not sure if there is a more ergonomic way to do it with the Python proto lib?

**Comment by erikbern** [comment]
> Mostly future proofing - the returned handle_metadata (previously object) can be any handle's proto object, but is currently a one_of that only has function as a valid field. When/if we add handle_metadata for any other object, this will make it automatically pick the "one_of" that is set. Not sure if there is a more ergonomic way to do it with the Python proto lib?

Got it – I guess this is a pure protobuf thing that has nothing to do with objects/functions – maybe we should break it out into a utility function? I.e.

```python
def get_oneof(message: google.protobuf.Message, oneof_attr: str):
    attr: str = message.OneOf(oneof_attr)
    return getattr(message, attr)
```

File: modal_utils/grpc_utils.py [hunk 1]
```diff
@@ -284,3 +285,11 @@ async def fallback(self, stream) -> None:
 
     cls.__abstractmethods__ = frozenset()
     return cls
+
+
+def get_proto_oneof(message: Message, oneof_group: str) -> Optional[Message]:
```
Comment (erikbern) [inline modal_utils/grpc_utils.py:290]: awesome thanks!

**Pull request merged**